neuralpredictors/layers/readouts/multi_readout.py

```diff
@@ -1,107 +1,147 @@
-class MultiReadout(Readout, ModuleDict):
+import torch
+from .base import *
+from .point_pooled import *
+from .gaussian import *
+from .factorized import *
+from .attention import *
+from .pyramid import *
+
+
+class MultiReadoutBase(torch.nn.ModuleDict):
+    """
+    Base class for MultiReadouts. It is a dictionary of data keys and readouts to the corresponding datasets.
+    If parameter-sharing between the readouts is desired, refer to MultiReadoutSharedParametersBase.
+
+    Args:
+        in_shape_dict (dict): dictionary of data_key and the corresponding dataset's shape as an output of the core.
+        n_neurons_dict (dict): dictionary of data_key and the corresponding dataset's number of neurons
+        base_readout (torch.nn.Module): base readout class. If None, self._base_readout must be set manually in the inheriting class's definition
+        mean_activity_dict (dict): dictionary of data_key and the corresponding dataset's mean responses. Used to initialize the readout bias with.
+                                   If None, the bias is initialized with 0.
+        clone_readout (bool): whether to clone the first data_key's readout to all other readouts, only allowing for a scale and offset.
+                              This is a rather simple method to enforce parameter-sharing between readouts. For more sophisticated methods,
+                              refer to MultiReadoutSharedParametersBase
+        gamma_readout (float): regularization strength
+        **kwargs:
+    """
+
     _base_readout = None
 
-    def __init__(self, in_shape, loaders, gamma_readout, clone_readout=False, **kwargs):
+    def __init__(
+        self,
+        in_shape_dict,
+        n_neurons_dict,
+        base_readout=None,
+        mean_activity_dict=None,
+        clone_readout=False,
+        gamma_readout=1.0,
```

Comment:
Several point here:

1. Why is there the assumption that `gamma_readout` applies to any readout?
2. why the inconsistency between `gamma_readout` and `reg_weight`? I guess due to backwards compatibility?

Suggestion: Let's leave any regulariziation-related argument in the kwargs. It can easily happen that a readout has multiple regularization each with their own specific regularization coefficient. Putting this argument in the kwargs generalizes the input arguments and also preserves backwards compatibility.

Reply:
The reason behind the new name is just that gamma_raedout is a terrible name and I feel like now is the only time to change it. I like your comment about leaving regularization to the kwargs, what do other people think about it?

Reply:
@eywalker @KonstantinWilleke what do you guys think?

Reply:
Nice, I like the idea a lot andI agree with leaving it in the `kwargs`. And it is a good name to change the name to be more descriptive. I'd advocate that we rename `gamma_readout` into `gamma_features`, since this is what is usually being regularized in all of our readouts. (Except Factorized, but we can tackle this later on).